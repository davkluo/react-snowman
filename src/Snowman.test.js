import { render, fireEvent } from "@testing-library/react";
import Snowman from './Snowman';

/**
 * Use fireEvent to simulate guessing letters
 *
 * @param {} container - HTML that contains letter buttons
 * @param {String} ltrs - Letters to guess
 */

function guessLetters(container, ltrs) {
  for (let ltr of ltrs) {
    const ltrBtn = container.querySelector(`button[value="${ltr}"]`)
    fireEvent.click(ltrBtn);
  }
}

it('renders without crashing', function() {
  render(<Snowman />);
});

it('matches snapshot', function() {
  const { container } = render(<Snowman />);
  expect(container).toMatchSnapshot();
})

it('successfully detects game over', function() {
  const { container } = render(<Snowman words={['z']}/>);

  guessLetters(container, 'abcdefg'); // Guess 7 times

  // Check image is still there
  const img = container.querySelector('.Snowman img');
  expect(container).toContainElement(img);

  // Check buttons are gone
  expect(container.querySelectorAll('.Snowman-ltr-btn').length).toEqual(0);

  // Check for lose message and correct word
  expect(container).toContainHTML('You lose');
  expect(container).toContainHTML('Correct word was: z');
})