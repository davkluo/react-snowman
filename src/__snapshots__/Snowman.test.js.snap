// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="0"
      src="0.png"
    />
    <p
      class="Snowman-num-wrong"
    >
      Number wrong: 
      0
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
      _
    </p>
    <p>
      <button
        class="Snowman-ltr-btn"
        value="a"
      >
        a
      </button>
      <button
        class="Snowman-ltr-btn"
        value="b"
      >
        b
      </button>
      <button
        class="Snowman-ltr-btn"
        value="c"
      >
        c
      </button>
      <button
        class="Snowman-ltr-btn"
        value="d"
      >
        d
      </button>
      <button
        class="Snowman-ltr-btn"
        value="e"
      >
        e
      </button>
      <button
        class="Snowman-ltr-btn"
        value="f"
      >
        f
      </button>
      <button
        class="Snowman-ltr-btn"
        value="g"
      >
        g
      </button>
      <button
        class="Snowman-ltr-btn"
        value="h"
      >
        h
      </button>
      <button
        class="Snowman-ltr-btn"
        value="i"
      >
        i
      </button>
      <button
        class="Snowman-ltr-btn"
        value="j"
      >
        j
      </button>
      <button
        class="Snowman-ltr-btn"
        value="k"
      >
        k
      </button>
      <button
        class="Snowman-ltr-btn"
        value="l"
      >
        l
      </button>
      <button
        class="Snowman-ltr-btn"
        value="m"
      >
        m
      </button>
      <button
        class="Snowman-ltr-btn"
        value="n"
      >
        n
      </button>
      <button
        class="Snowman-ltr-btn"
        value="o"
      >
        o
      </button>
      <button
        class="Snowman-ltr-btn"
        value="p"
      >
        p
      </button>
      <button
        class="Snowman-ltr-btn"
        value="q"
      >
        q
      </button>
      <button
        class="Snowman-ltr-btn"
        value="r"
      >
        r
      </button>
      <button
        class="Snowman-ltr-btn"
        value="s"
      >
        s
      </button>
      <button
        class="Snowman-ltr-btn"
        value="t"
      >
        t
      </button>
      <button
        class="Snowman-ltr-btn"
        value="u"
      >
        u
      </button>
      <button
        class="Snowman-ltr-btn"
        value="v"
      >
        v
      </button>
      <button
        class="Snowman-ltr-btn"
        value="w"
      >
        w
      </button>
      <button
        class="Snowman-ltr-btn"
        value="x"
      >
        x
      </button>
      <button
        class="Snowman-ltr-btn"
        value="y"
      >
        y
      </button>
      <button
        class="Snowman-ltr-btn"
        value="z"
      >
        z
      </button>
    </p>
  </div>
</div>
`;
